lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -369,21 +375,21 @@ data ApiAddress (n :: NetworkDiscriminant) = ApiAddress
     , state :: !(ApiT AddressState)
     } deriving (Eq, Generic, Show)
 
-newtype ApiAddressDerivationPath = ApiAddressDerivationPath
-    { unApiAddressDerivationPath :: NonEmpty ApiAddressDerivationSegment
+newtype ApiDerivationPath = ApiDerivationPath
+    { unApiDerivationPath :: NonEmpty ApiDerivationSegment
     } deriving (Eq, Generic, Show)
 
-data ApiAddressDerivationSegment = ApiAddressDerivationSegment
-    { derivationIndex :: !ApiRelativeAddressIndex
-    , derivationType :: !ApiAddressDerivationType
+data ApiDerivationSegment = ApiDerivationSegment
```

Comment:
I want to use this data type in the `Cardano.Wallet`, which must not use API types. Because this is similar to 

```hs
newtype Index (derivationType :: DerivationType) (level :: Depth) = Index
    { getIndex :: Word32 }
```
which I can't use as in a homogenous List easily. `ApiAddressDerivationSegment` would basically be non-typelevel version of `Index`.

Reply:
Or... we could define a toText instance for Index

Reply:
What about using the `DerivationIndex` from the `Primitive.Types` instead ?

Reply:
That doesn't preserve the derivation type, don't we need that?

Reply:
When considering all indexes as a list (or path):

- the depth matters not because what the `Depth` captures is actually the position of the index in that list. 
- the derivationType is captured by representing indexes as plain Word32. The `Soft` / `Hardened` notation is for easing human-readability but in the end, a soft index is simply a value < 2^31, whereas a "hardened" index is simply a value >= 2^31. Therefore, instead of representing indexes as derivationType + relative index within 0 and 2^31, we can represent them as just an index between 0 and 2^32, which is what `DerivationIndex` does.

Reply:
I should probably add these bits as comments.